crates/compiler/builtins/bitcode/src/lib.rs

```diff
@@ -21,45 +19,71 @@ pub fn host_wasm_tempfile() -> std::io::Result<NamedTempFile> {
     Ok(tempfile)
 }
 
-#[cfg(unix)]
-fn host_unix_tempfile() -> std::io::Result<NamedTempFile> {
-    let tempfile = tempfile::Builder::new()
-        .prefix("host_bitcode")
-        .suffix(".o")
-        .rand_bytes(8)
-        .tempfile()?;
+fn current_host_tempfile() -> &'static std::io::Result<NamedTempFile> {
+    use std::sync::OnceLock;
 
-    std::fs::write(tempfile.path(), HOST_UNIX)?;
+    static TMP: OnceLock<std::io::Result<NamedTempFile>> = OnceLock::new();
 
-    Ok(tempfile)
-}
+    fn helper() -> std::io::Result<NamedTempFile> {
+        let tempfile = tempfile::Builder::new()
+            .prefix("host_bitcode")
+            .suffix(".o")
+            .rand_bytes(8)
+            .tempfile()?;
 
-#[cfg(windows)]
-fn host_windows_tempfile() -> std::io::Result<NamedTempFile> {
-    let tempfile = tempfile::Builder::new()
-        .prefix("host_bitcode")
-        .suffix(".obj")
-        .rand_bytes(8)
-        .tempfile()?;
+        std::fs::write(tempfile.path(), HOST_UNIX)?;
 
-    std::fs::write(tempfile.path(), HOST_WINDOWS)?;
+        Ok(tempfile)
+    }
 
-    Ok(tempfile)
+    TMP.get_or_init(helper)
 }
 
-pub fn host_tempfile() -> std::io::Result<NamedTempFile> {
-    #[cfg(unix)]
-    {
-        host_unix_tempfile()
-    }
+fn host_windows_tempfile() -> &'static std::io::Result<NamedTempFile> {
+    use std::sync::OnceLock;
+
+    static TMP: OnceLock<std::io::Result<NamedTempFile>> = OnceLock::new();
 
-    #[cfg(windows)]
-    {
-        host_windows_tempfile()
+    fn helper() -> std::io::Result<NamedTempFile> {
+        let tempfile = tempfile::Builder::new()
+            .prefix("host_bitcode")
+            .suffix(".obj")
+            .rand_bytes(8)
+            .tempfile()?;
+
+        std::fs::write(tempfile.path(), HOST_WINDOWS)?;
+
+        Ok(tempfile)
     }
 
-    #[cfg(not(any(windows, unix)))]
-    {
-        unreachable!()
+    TMP.get_or_init(helper)
+}
+
+pub fn host_tempfile(target: &target_lexicon::Triple) -> &'static std::io::Result<NamedTempFile> {
+    use target_lexicon::Triple;
+
+    match target {
+        Triple {
+            // architecture: target_lexicon::Architecture::X86_64,
+            // operating_system: target_lexicon::OperatingSystem::Linux,
```

Comment:
in practice the "unix" host file is the file compiled for this host. That means that we need to be pretty vague in this pattern to not cause regressions. Eventually we should generate the builtins for more targets, and be more specific here.